This should really be done in bam's NewSettings.

Lubomir Rintel <lkundrak@v3.sk>

diff -urp teeworlds-0.5.0-src/default.bam teeworlds-0.5.0-src.fixed/default.bam
--- teeworlds-0.5.0-src/default.bam	2009-01-17 14:20:53.000000000 +0100
+++ teeworlds-0.5.0-src.fixed/default.bam	2009-01-17 14:18:23.000000000 +0100
@@ -106,6 +109,9 @@ end
 function build(settings)
 	--settings.objdir = Path("objs")
 	settings.cc.Output = Intermediate_Output
+	settings.cc.flags:Add(os.getenv ('CFLAGS'))
+	settings.cc.flags:Add(os.getenv ('CPPFLAGS'))
+	settings.cc.flags:Add(os.getenv ('LDFLAGS'))
 
 	if config.compiler.value == "cl" then
 		settings.cc.flags:Add("/wd4244")
